docs/requirements.txt

```diff
@@ -1,3 +1,2 @@
-m2r
```

Comment:
Leaving a trail in case someone comes back to the PR later.

We opted to remove `m2r` because the dependency does _not_ install with Python 3.10.  Here's the error:

```
  Using cached m2r-0.1.0.tar.gz (10 kB)
    ERROR: Command errored out with exit status 1:
     command: /Users/michael/dev/pyteal/.venv/bin/python3 -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/m6/4mf0p_y17yn_sx2gt7fpcj980000gp/T/pip-install-s36zhju9/m2r_0fbb41fce6504a278f0ec41cd3a97e49/setup.py'"'"'; __file__='"'"'/private/var/folders/m6/4mf0p_y17yn_sx2gt7fpcj980000gp/T/pip-install-s36zhju9/m2r_0fbb41fce6504a278f0ec41cd3a97e49/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /private/var/folders/m6/4mf0p_y17yn_sx2gt7fpcj980000gp/T/pip-pip-egg-info-sgk4t6mj
         cwd: /private/var/folders/m6/4mf0p_y17yn_sx2gt7fpcj980000gp/T/pip-install-s36zhju9/m2r_0fbb41fce6504a278f0ec41cd3a97e49/
    Complete output (7 lines):
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/private/var/folders/m6/4mf0p_y17yn_sx2gt7fpcj980000gp/T/pip-install-s36zhju9/m2r_0fbb41fce6504a278f0ec41cd3a97e49/setup.py", line 13, in <module>
        from m2r import parse_from_file
      File "/private/var/folders/m6/4mf0p_y17yn_sx2gt7fpcj980000gp/T/pip-install-s36zhju9/m2r_0fbb41fce6504a278f0ec41cd3a97e49/m2r.py", line 22, in <module>
        class RestBlockGrammar(mistune.BlockGrammar):
    AttributeError: module 'mistune' has no attribute 'BlockGrammar'
```

Additionally, the dependency is deprecated and replaced by `m2r2`.  We confirmed the replacement dependency works with `docs/`, but opted to remove it because it's unused.